pkg/didcomm/common/service/event.go

```diff
@@ -49,24 +49,24 @@ type DIDCommAction struct {
 	//
 	// Supported protocols
 	//   - DID Exchange :  didexchange.DIDExchange
-	ProtocolName string
+	ProtocolName string `json:",omitempty"`
```

Comment:
why not do similar to service.StateMsg?

note:

* service.StateMsg.Type will be serialized as an int
* service.StateMsg.Msg will leak internal details from service.DIDCommMsgMap
* not sure if all service.StateMsg.EventProperties will serialize properly

Reply:
I added `omitempty` everywhere. We do not need empty values, right?

I would not say that is a leak of internal details. 
We actually provide the same functionality. We just replacing channels with WebSocket.
Currently, EventProperties are working fine. 
I can create different structures for that.
Just let me know.

Reply:
About properties, I am pretty sure they should be serialized and exposed as well. But I understand that because of interface it cannot be safe. What we can do here is to add a comment that properties should be serialized otherwise event can be skipped for the WebSocket.

Reply:
@soluchok 

> We actually provide the same functionality. We just replacing channels with WebSocket.

It's not the same. `service.DIDCommMsgMap` is actually exposed as the interface `service.DIDCommMsg` in the framework. From there users cannot access the metadata (unless they follow your example and start casting it to DIDCommMsgMap...).

> About properties, I am pretty sure they should be serialized and exposed as well. But I understand that because of interface it cannot be safe. What we can do here is to add a comment that properties should be serialized otherwise event can be skipped for the WebSocket.

I'm not sure what you mean regarding safety, but I'm sure, as an example, the following two properties won't be serialized: https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/protocol/didexchange/event.go#L19

Reply:
Done! no more casting